node-scripts/spec/pipeline3/scripts.spec.js

```diff
@@ -517,4 +414,92 @@ describe("Scripts", function() {
         "\n"
     );
   });
+
+  it("should generate 'replace environment variable' script", function() {
+    var envVar = "KEY";
+    var value = "env.value";
+    var filename = ".env";
+
+    expect(scripts.writeProperty(envVar, value, filename)).toEqual(
+      'if ! grep -R "^[#]*s*KEY.*" .env > /dev/null; then\n' +
+        "\techo \"APPENDING because 'KEY' not found\"\n" +
+        '\techo "KEY=env.value" >> .env\n' +
+        "else\n" +
+        "\techo \"SETTING because 'KEY' found already\"\n" +
+        '\tsed -i "s/^[#]*\\s*KEY.*/KEY=env.value/" .env\n' +
+        "fi\n"
+    );
+  });
+
+  it("should generate 'create distro.env file' script", function() {
+    var instanceDef = {
+      name: "hsc-dev",
+      uuid: "1-22-333",
+      scripts: [
+        {
+          type: "shell",
+          executionStage: "6",
+          conditions: ["data"],
+          value: "/a/script.sh"
+        }
+      ],
+      deployment: {
+        host: {
+          type: "ssh",
+          value: {
+            ip: "54.154.133.95",
+            user: "ec2-user",
+            port: "22"
+          }
+        },
+        type: "dockerCompose"
+      },
+      envVars: {
+        prop: "value",
+        prop2: "value2",
+        prop3: "value3"
+      }
+    };
+  });
+
+  it("should generate 'git clone' script", function() {
+    var gitUrl = "git@github:mekomsolutions/bahmni-docker.git";
+    var commitId = "12b9a94";
+
+    expect(scripts.gitClone(gitUrl, "/path/to/distro", commitId)).toEqual(
+      "git clone git@github:mekomsolutions/bahmni-docker.git /path/to/distro\n" +
+        "cd /path/to/distro && git checkout 12b9a94\n"
+    );
+  });
+  it("should generate 'distro.env file'", () => {
```

Comment:
Isn't this redundant with the one line 434?

Reply:
yes